web-api/src/ingestion/preprocess_document.rs

```diff
@@ -53,3 +55,21 @@ fn embed_with_summarizer(
     let embedding = embedder.run(&summary)?;
     Ok(vec![(snippet.clone(), embedding)])
 }
+
+fn embed_with_cutters(
+    embedder: &Embedder,
+    snippet: &DocumentSnippet,
+) -> Result<DocumentContent, Error> {
+    let mut snippets = Vec::new();
+    for split in cutters::cut(snippet, cutters::Language::English) {
+        let str = split.str.trim();
+        let snippet = DocumentSnippet::new(str, str.len())?;
+        let embedding = embedder.run(split.str)?;
+        snippets.push((snippet, embedding));
+    }
```

Comment:
:art: instead of a push-loop we could just iterate on the cutter values and collect afterwards. (if we stick to the push loop, then we should at least allocate snippets with the capacity to hold all splits from `cutters::cut()` since this would be the happy path of the loop.)
```suggestion
    let snippets = cutters::cut(snippet, cutters::Language::English)
        .into_iter()
        .map(|split| {
            let snippet = DocumentSnippet::new(split.str, snippet.len())?;
            let embedding = embedder.run(&snippet)?;
            Ok::<_, Error>((snippet, embedding))
        })
        .try_collect::<_, Vec<_>, _>()?;
```